core/src/main/java/com/scalar/db/api/ScanBuilder.java

```diff
@@ -514,54 +499,33 @@ public static class BuildableScanAllWithWhere
           Ordering<BuildableScanAllWithWhere>,
           Limit<BuildableScanAllWithWhere> {
 
-    protected final String namespaceName;
-    protected final String tableName;
-    @Nullable protected final ConditionalExpression condition;
-    protected final Set<Set<ConditionalExpression>> conjunctions = new HashSet<>();
-    protected final Set<Set<ConditionalExpression>> disjunctions = new HashSet<>();
-    protected final List<String> projections = new ArrayList<>();
-    protected final List<Scan.Ordering> orderings = new ArrayList<>();
-    protected int limit;
-    @Nullable protected com.scalar.db.api.Consistency consistency;
+    protected BuildableScanAll buildableScanAll;
```

Comment:
[trivial] `BuildableScanAll` is mutable, so you can also make this variable `final` and simply use it like this.

```java
    @Override
    public BuildableScanAllWithWhere projection(String projection) {
      buildableScanAll.projection(projection);
      return this;
    }
```

The current code can be used as-is if `BuildableScanAll` changes into immutable in the future, so I don't have a strong opinion on it , though.

Reply:
Thanks again. Definitely. Fixed them in 8320ae8 as you mentioned due to no plan for changing them to immutable so far.